include/partitioner/cell_storage.hpp

```diff
@@ -52,11 +52,9 @@ namespace detail
 template <storage::Ownership Ownership> class CellStorageImpl
 {
   public:
-    using ValueOffset = std::uint32_t;
-    using BoundaryOffset = std::uint32_t;
+    using ValueOffset = std::size_t;
+    using BoundaryOffset = std::size_t;
     using BoundarySize = std::uint32_t;
```

Comment:
OSM contains ~800 million ways, so 32 bit should be safe to use for boundary sizes..for now.